### PR TITLE
Do not use dict to store spec sections

### PR DESCRIPTION
**Pull request by nforro**
Sections in a spec file can be enclosed in conditions, which means they can be duplicate.
Use list of tuples instead of dict to store them.

### CODE REVIEW

File: rebasehelper/specfile.py [hunk 1]
```diff
@@ -147,13 +146,49 @@ def __init__(self, content):
...
         content = []
-        for header, section in six.iteritems(self.sections):
+        for header, section in self.sections:
             if header != '%package':
                 content.append(header + '\n')
             for line in section:
                 content.append(line + '\n')
         return ''.join(content)
 
+    def section(self, name):
```
Comment (FrNecas) [inline rebasehelper/specfile.py]: Shouldn't this rather be named `get_section`?

Comment (nforro) [reply rebasehelper/specfile.py]: It certainly could, but it is supposed to be a drop-in replacement for the current `spec_content.section[NAME]`.

File: rebasehelper/specfile.py [hunk 2]
```diff
@@ -147,13 +146,49 @@ def __init__(self, content):
...
+        """Replaces content of a section.
+
+        In case there are multiple sections with the same name, the first one is replaced.
+
+        Args:
+            name (str): Section name.
+            content (list): Section content as a list of lines.
+
+        Returns:
+            bool: False if section was not found else True
```
Comment (FrNecas) [inline rebasehelper/specfile.py]: Just a nitpick - there should be a dot at the end of the line

File: rebasehelper/specfile.py (hunk 1 above)
Comment (FrNecas) [reply rebasehelper/specfile.py]: You are right that, moreover `get_section` would be a bit too long with our use-case in mind

**Pull request merged**